### PR TITLE
Fix gather kernel bug for handling 5 and 6 dimensions

### PR DESCRIPTION
**Pull request by clee30**
### Details:
The calculation to generate INDICES_INDEX_ORDER is wrong.

For 5 dimension, INDICES_INDEX_ORDER should be
b, f, 0, z, 0 or b, f, 0, 0, z

For 6 dimension, INDICES_INDEX_ORDER should be
b, f, 0, w, z, 0 or b, f, 0, w, 0,z

To simplify the generation, generate b,f,0,z,0 for 5 dims and b,f,0,w,z,0 for 6 dimsm

### Tickets:
 - [*CVS-150766*](https://jira.devtools.intel.com/browse/CVS-150766)

### CODE REVIEW

**Comment by vladimir-paramuzov** [comment]
build_jenkins

File: src/plugins/intel_gpu/src/kernel_selector/kernels/gather/gather_kernel_ref.cpp [hunk 1]
```diff
@@ -186,9 +186,21 @@ static std::string GetIndicesIdxOrder(const gather_params& params, size_t axis,
 
     size_t indices_dims_num = GetNonEmptyDimsNumber(params.inputs[1]);
 
-    // Shift indices of Gather indices input related to output dims
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/kernel_selector/kernels/gather/gather_kernel_ref.cpp:189]: [random spot] Please add functional test case that covers fixed issue here: https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/tests/functional/shared_tests_instances/single_layer_tests/gather.cpp

Comment (clee30) [reply src/plugins/intel_gpu/src/kernel_selector/kernels/gather/gather_kernel_ref.cpp:189]: Added the earlier failure tensorflow_tests that uncover the bug.  Should I still need to add more test in gather.cpp?

**Comment by vladimir-paramuzov** [comment]
build_jenkins

**Review by rkazants — APPROVED** [review]
From TF perspective, LGTM

**Comment by vladimir-paramuzov** [comment]
@clee30 Looks like some unit tests fail:
```
[----------] 1 test from gather8_gpu_fp16
[ RUN      ] gather8_gpu_fp16.d323_axisY_bdim_m1
src/plugins/intel_gpu/tests/unit/test_cases/gather_gpu_test.cpp:454: Failure
Expected equality of these values:
  expected_results[i]
    Which is: 27
  half_to_float(output_ptr[i])
    Which is: 31
[  FAILED  ] gather8_gpu_fp16.d323_axisY_bdim_m1 (457 ms)
[----------] 1 test from gather8_gpu_fp16 (457 ms total)

[----------] 6 tests from gather7_gpu_fp16
[ RUN      ] gather7_gpu_fp16.d222_axisX_bdim_m1
** Segmentation fault
...
```

**Comment by clee30** [comment]
> @clee30 Looks like some unit tests fail:
> 
> ```
> [----------] 1 test from gather8_gpu_fp16
> [ RUN      ] gather8_gpu_fp16.d323_axisY_bdim_m1
> src/plugins/intel_gpu/tests/unit/test_cases/gather_gpu_test.cpp:454: Failure
> Expected equality of these values:
>   expected_results[i]
>     Which is: 27
>   half_to_float(output_ptr[i])
>     Which is: 31
> [  FAILED  ] gather8_gpu_fp16.d323_axisY_bdim_m1 (457 ms)
> [----------] 1 test from gather8_gpu_fp16 (457 ms total)
> 
> [----------] 6 tests from gather7_gpu_fp16
> [ RUN      ] gather7_gpu_fp16.d222_axisX_bdim_m1
> ** Segmentation fault
> ...
> ```

Thanks.   The gather unit tests are passing.  Will push the fix in soon.

**Comment by clee30** [comment]
> @clee30 Looks like some unit tests fail:
> 
> ```
> [----------] 1 test from gather8_gpu_fp16
> [ RUN      ] gather8_gpu_fp16.d323_axisY_bdim_m1
> src/plugins/intel_gpu/tests/unit/test_cases/gather_gpu_test.cpp:454: Failure
> Expected equality of these values:
>   expected_results[i]
>     Which is: 27
>   half_to_float(output_ptr[i])
>     Which is: 31
> [  FAILED  ] gather8_gpu_fp16.d323_axisY_bdim_m1 (457 ms)
> [----------] 1 test from gather8_gpu_fp16 (457 ms total)
> 
> [----------] 6 tests from gather7_gpu_fp16
> [ RUN      ] gather7_gpu_fp16.d222_axisX_bdim_m1
> ** Segmentation fault
> ...
> ```

Fixed.

**Comment by vladimir-paramuzov** [comment]
build_jenkins

**Pull request merged**